satpy/tests/reader_tests/test_seviri_l2_grib.py

```diff
@@ -0,0 +1,199 @@
+                        # Asserts that calculate_area_extent has been called with the correct arguments
+                        name, args, kwargs = cae.mock_calls[0]
+                        self.assertEqual(args[0]['resolution'], 400.)
+                        # Asserts that get_area_definition has been called with the correct arguments
+                        name, args, kwargs = gad.mock_calls[0]
+                        self.assertEqual(args[0], expected_pdict)
+                        # The second argument must be the return result of calculate_area_extent
+                        self.assertEqual(args[1]._extract_mock_name(), 'calculate_area_extent()')
+
+                # Restarts the id generator and clears the call history
+                fake_gid_generator = (i for i in FAKE_GID)
+                self.ec_.codes_grib_new_from_file.side_effect = lambda fh: next(fake_gid_generator)
+                self.ec_.codes_grib_new_from_file.reset_mock()
+                self.ec_.codes_release.reset_mock()
+
+                # Checks the correct execution of the get_dataset function with a valid parameter_number
+                valid_dataset = self.reader.get_dataset(None, {'parameter_number': 30})
+                # Checks that the dataset has been created as a DataArray object
+                self.assertEqual(valid_dataset._extract_mock_name(), 'xr.DataArray()')
+                # Checks that codes_release has been called after each codes_grib_new_from_file call
+                self.assertEqual(self.ec_.codes_grib_new_from_file.call_count,
+                                 self.ec_.codes_release.call_count)
+
+                # Restarts the id generator and clears the call history
+                fake_gid_generator = (i for i in FAKE_GID)
+                self.ec_.codes_grib_new_from_file.side_effect = lambda fh: next(fake_gid_generator)
+                self.ec_.codes_grib_new_from_file.reset_mock()
+                self.ec_.codes_release.reset_mock()
+
+                # Checks the correct execution of the get_dataset function with an invalid parameter_number
+                invalid_dataset = self.reader.get_dataset(None, {'parameter_number': 50})
+                # Checks that the function returns None
+                self.assertEqual(invalid_dataset, None)
+                # Checks that codes_release has been called after each codes_grib_new_from_file call
+                # (except after the last one which has returned a None)
+                self.assertEqual(self.ec_.codes_grib_new_from_file.call_count,
+                                 self.ec_.codes_release.call_count + 1)
+
+
+def suite():
+    """Build the test suite for test_scene."""
+    loader = unittest.TestLoader()
+    mysuite = unittest.TestSuite()
+
+    mysuite.addTest(loader.loadTestsFromTestCase(Test_SeviriL2GribFileHandler))
+
+    return mysuite
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
These can be removed upon merge with master

Reply:
removed.